include/ada/url-inl.h

```diff
@@ -133,7 +133,7 @@ inline void url::update_base_search(std::string_view input,
 }
 
 inline void url::update_base_search(std::optional<std::string> input) {
-  query = input;
+  query = std::move(input);
```

Comment:
```suggestion
  query = input;
```

There is no need if you pass an r-value as I suggest.

Reply:
I am wrong. You still need the std::move there. Sorry about that.